mathesar/state/base.py

```diff
@@ -2,6 +2,40 @@
 from mathesar.state.metadata import reset_cached_metadata, get_cached_metadata
 from mathesar.state.cached_property import clear_cached_property_cache
 
+CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID = {}
+GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID = {}
+
+
+def get_preview_columnId_for_tableId(tableId):
+    if tableId in CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID:
+        return CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID[tableId]
+    else:
+        return None
+
+
+def set_preview_columnId_for_tableId(tableId, previewColumnId):
+    CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID[tableId] = previewColumnId
+    GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID[previewColumnId] = tableId
+
+
+def remove_preview_columnId_from_cache(columnId):
+    if columnId in GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID:
+        tableId = GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID[columnId]
+        GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID.pop(columnId, None)
+        CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID.pop(tableId, None)
+
+
+def remove_tableId_from_cache(tableId):
+    if tableId in CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID:
+        columnId = CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID[tableId]
+        GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID.pop(columnId, None)
+        CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID.pop(tableId, None)
+
+
+def clear_preview_cache():
+    CACHED_PREVIEW_COLUMN_ID_FOR_TABLE_ID.clear()
+    GET_TABLE_ID_FROM_PREVIEW_COLUMN_ID.clear()
+
```

Comment:
This file is not the appropriate place for this logic. E.g. routines related to `clear_dj_cache`, and it itself, aren't defined in this file.